R/ShinyArchRExports.R

```diff
@@ -0,0 +1,620 @@
+# Functions for exporting an ArchR-based Shiny app -----------------------------------------------------------
+#'
+#' Export a Shiny App based on an ArchRProj
+#' 
+#' Generate all files required for an autonomous Shiny app to display browser tracks and embeddings.
+#'
+#' @param ArchRProj An `ArchRProject` object.
+#' @param outputDir The name (not the path!) of the directory for the Shiny App files. This will become a sub-directory of the ArchRProj output directory
+#' given by `getOutputDirectory(ArchRProj)`.
+#' @param subOutputDir Where data to upload to Shinyapps.io will be stored. Defaults to `inputData`.
+#' @param groupBy The name of the column in `cellColData` to use for grouping cells together for generating BigWig-style sequencing tracks. 
+#' Only one cell grouping is allowed.
+#' @param cellColEmbeddings A character vector of columns in `cellColData` to plot as part of the Shiny app. No default is provided so this must be set.
+#' For ex. `c("Sample","Clusters","TSSEnrichment","nFrags")`.
+#' @param embedding The name of the embedding from `ArchRProj` to be used for plotting embeddings in the Shiny app.
+#' @param tileSize The numeric width of the tile/bin in basepairs for plotting ATAC-seq signal tracks. All insertions in a single bin will be summed.
+#' @param force A boolean value that indicates whether to overwrite any relevant files during the `exportShinyArchR()` process.
+#' @param threads The number of threads to use for parallel execution.
+#' @param logFile The path to a file to be used for logging ArchR output.
+#' 
+#' @examples
+#' 
+#' proj <- getTestProject(version = 2)
+#' proj@geneAnnotation$genes <- proj@geneAnnotation$genes[which(proj@geneAnnotation$genes$symbol %in% c("CD14","CD3D","MS4A1","CD74"))]
+#' proj <- addGeneScoreMatrix(input = proj, force = TRUE)
+#' ArchR:::.dropGroupsFromArrow(ArrowFile = getArrowFiles(proj)[1], dropGroups = c("GeneIntegrationMatrix","MotifMatrix"))
+#' proj <- addImputeWeights(proj)
+#'
+#' ArchR:::exportShinyArchR(ArchRProj = proj,
+#'                  mainDir = "Shiny",
+#'                  subOutDir = "inputData",
+#'                  savedArchRProjFile = "Save-ArchR-Project.rds",
+#'                  groupBy = "Clusters",
+#'                  cellColEmbeddings = "Clusters",
+#'                  embedding = "UMAP",
+#'                  tileSize = 100,
+#'                  force = FALSE,
+#'                  threads = getArchRThreads(),
+#'                  logFile = createLogFile("exportShinyArchR"))
+#' 
+#' @export
+exportShinyArchR <- function(
+  ArchRProj = NULL,
+  mainDir = "Shiny",
+  subOutDir = "inputData",
+  # ArchRProjFile = "Save-ArchRProjShiny",
+  savedArchRProjFile = "Save-ArchR-Project.rds",
+  groupBy = "Clusters",
+  cellColEmbeddings = "Clusters",
+  embedding = "UMAP",
+  tileSize = 100,
+  force = FALSE,
+  threads = getArchRThreads(),
+  logFile = createLogFile("exportShinyArchR")
+){
+  
+  options(warn=-1)
```

Comment:
not good coding practice. If there are specific warnings you want to ignore, I did write a function to do that. let me know.

Reply:
`options(warn=-1)` no longer in [2987355](https://github.com/paupaiz/ArchR/commit/2987355c6505adeec2a75203650b1e1aee064866)